src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -489,6 +489,115 @@ void regclass_graph_Model(py::module m) {
 
     model.def("validate_nodes_and_infer_types", &ov::Model::validate_nodes_and_infer_types);
 
+    model.def(
+        "reshape",
+        [](ov::Model& self, const py::object& shapes) {
+        std::map<ov::Output<ov::Node>, ov::PartialShape> new_shapes;
+        if (py::isinstance<py::dict>(shapes)) {
+            py::dict shapes_dict = shapes.cast<py::dict>();
+            for (auto& item : shapes_dict) {
+                new_shapes.emplace(output_from_handle(self, item.first), partial_shape_from_handle(item.second));
+            }
```

Comment:
You repeat here what [this](https://github.com/openvinotoolkit/openvino/blob/fab9a560f9b9bd06aa306be02a072ed644b0a61f/src/bindings/python/src/pyopenvino/graph/model.cpp#L662) reshape overload does